lib/streamlit/UploadedFileManager.py

```diff
@@ -53,39 +44,45 @@ class UploadedFileManager(object):
     """
 
     def __init__(self):
-        self._files = {}
+        self._files_by_id = {}  # type: Dict[Tuple[str, str], UploadedFileList]
         self._files_lock = threading.Lock()
-        self.on_file_added = Signal(
-            doc="""Emitted when a file is added to the manager.
+        self.on_files_added = Signal(
+            doc="""Emitted when a file list is added to the manager.
```

Comment:
Maybe there is away to rephrase this to something like `when an addition is made to the file manager` so it is independent of whether is a list or not

Reply:
The argument is *always* a file list, in this case. (Even single files must be wrapped in a list.)

Reply:
Ok. Seems redundant to say it is a list here if it is always a list and should be clear from the function signature, but more of a personal thought.